src/lm_saes/sae_training.py

```diff
@@ -325,4 +341,76 @@ def prune_sae(
         )
         sae.save_pretrained(path)
         
-    return sae
\ No newline at end of file
+    return sae
+
+@torch.no_grad()
+def init_sae_on_dataset(
```

Comment:
It seems better to accept an SAE as input and initialize its parameter, rather than directly create SAEs in this function. Is there some technical reason for an out-place initialization?